GiEnJul/ClientApp/src/components/registerGiver/ContactInfo.tsx

```diff
@@ -1,103 +1,153 @@
+import { Button, Grid, Container } from "@material-ui/core";
+import useStyles from "./Styles";
+import InputValidator from "../../components/InputFields/Validators/InputValidator";
+import {
+  isNotNull,
+  isPhoneNumber,
+  isEmail,
+} from "../../components/InputFields/Validators/Validators";
+import { useState } from "react";
+import IGiverFormData from "./IGiverFormData";
 
 type Props = {
-    nextStep: () => void,
-    prevStep: () => void,
-    handlefullnameChange: (event: React.ChangeEvent<HTMLInputElement>) => void,
-    handleEmailChange: (event: React.ChangeEvent<HTMLInputElement>) => void,
-    handleTlfChange: (event: React.ChangeEvent<HTMLInputElement>) => void,
-    values: { location?: string; fullname?: string; email?: string; phoneNumber?: string; maxRecievers?: number; familyType?: string; },
-}
+  nextStep: (event: React.FormEvent) => void;
+  prevStep: (event: React.FormEvent) => void;
+  handlefullnameChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleEmailChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleTlfChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  values: IGiverFormData;
+};
 
-const ContactInfo: React.FC<Props> = ({ nextStep, prevStep, handlefullnameChange, handleEmailChange, handleTlfChange, values }) => {
+type keyValue = {
+  [key: string]: boolean;
+};
 
-    const [viewErrorTrigger, setViewErrorTrigger] = useState(0);
+const initIsValidsState: keyValue = {
+  isNotEmptyFullName: false,
+  isNotEmptyEmail: false,
+  isValidEmail: false,
+  isNotEmptyPhone: false,
+  isValidPhone: false,
+};
 
-    const [isNotEmptyFullName, setIsNotEmptyFullName] = useState(false);
-    const [isNotEmptyEmail, setIsNotEmptyEmail] = useState(false);
-    const [isValidEmail, setIsValidEmail] = useState(false);
-    const [isNotEmptyPhone, setIsNotEmptyPhone] = useState(false);
-    const [isValidPhone, setIsValidPhone] = useState(false);
+const initState = {
+  viewErrorTrigger: 0,
+};
 
-    const Continue = (e: any) => {
-        e.preventDefault();
-        if (!(isNotEmptyFullName && isNotEmptyEmail && isValidEmail && isNotEmptyPhone && isValidPhone)) {
-            setViewErrorTrigger(v => v + 1);
-            return;
-        }
-        nextStep();
-    }
+const ContactInfo: React.FC<Props> = ({
+  nextStep,
+  prevStep,
+  handlefullnameChange,
+  handleEmailChange,
+  handleTlfChange,
+  values,
+}) => {
+  const [state, setState] = useState(initState);
+  const [isValidsState, setIsValidsState] = useState(initIsValidsState);
 
-    const Previous = (e: any) => {
-        e.preventDefault();
-        prevStep();
+  const extendedNextStep = (e: React.FormEvent) => {
+    for (let isValid in isValidsState) {
+      if (isValidsState[isValid]) continue;
+      return setState((prev) => {
+        return { ...prev, viewErrorTrigger: prev.viewErrorTrigger + 1 };
+      });
     }
+    nextStep(e);
+  };
+
+  const getValiditySetter = (target: string) => {
```

Comment:
While its nice to send confirmation that something is or isnt valid, for better flow in the form usually we want to make sure that validation happens when we need something from the user before moving to the next step. This function to me sounds like at the end before the user submits its form, thats when you'll let them know that something is wrong. Imagine that scenario. Just something to think about.